src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -156,9 +156,14 @@ export class Selection {
      */
     public isSelected(item: any): boolean {
         if (this._selectionType === SelectionType.Single) {
-            return this.currentSingle === item;
+            return (this.currentSingle) ? this._items.trackBy(0, this.currentSingle) === this._items.trackBy(0, item) :
+                                          false;
         } else if (this._selectionType === SelectionType.Multi) {
-            return this.current.indexOf(item) >= 0;
+            return this.current
+                       .filter((selected: any) => {
+                           return this._items.trackBy(0, selected) === this._items.trackBy(0, item);
+                       })
+                       .length > 0;
```

Comment:
I have a slight preference for using `findIndex` instead of `filter`, so that the function will return upon finding a match as opposed to continue going through the rest of the array.
```
return this.current.findIndex((selected: any) => {
    return this._items.trackBy(0, selected) === this._items.trackBy(0, item);
}) >= 0;
```